purchases/src/main/kotlin/com/revenuecat/purchases/common/events/EventsManager.kt

```diff
@@ -0,0 +1,187 @@
+            }
+
+            val storedEventsWithNullValues = getStoredEvents()
+            val storedEvents = storedEventsWithNullValues.filterNotNull()
+
+            if (storedEvents.isEmpty()) {
+                verboseLog("No new events to sync.")
+                flushInProgress = false
+                return@enqueue
+            }
+
+            verboseLog("New event flush: posting ${storedEvents.size} events.")
+            postEvents(
+                EventRequest(storedEvents),
+                {
+                    verboseLog("New event flush: success.")
+                    enqueue {
+                        fileHelper.clear(storedEventsWithNullValues.size)
+                        flushInProgress = false
+                    }
+                },
+                { error, shouldMarkAsSynced ->
+                    errorLog("New event flush error: $error.")
+                    enqueue {
+                        if (shouldMarkAsSynced) {
+                            fileHelper.clear(storedEventsWithNullValues.size)
+                        }
+                        flushInProgress = false
+                    }
+                },
+            )
+        }
+    }
+
+    private fun flushLegacyEvents() {
+        enqueue {
+            val storedLegacyEventsWithNullValues = getLegacyStoredEvents()
+            val storedLegacyEvents = storedLegacyEventsWithNullValues.filterNotNull()
+
+            if (storedLegacyEvents.isEmpty()) {
+                verboseLog("No legacy events to sync. Skipping legacy flush.")
+                return@enqueue
+            }
+
+            verboseLog("Legacy event flush: posting ${storedLegacyEvents.size} events.")
+            postEvents(
+                EventRequest(storedLegacyEvents),
+                {
+                    verboseLog("Legacy event flush: success.")
+                    enqueue { legacyEventsFileHelper.clear(storedLegacyEventsWithNullValues.size) }
+                },
+                { error, shouldMarkAsSynced ->
+                    errorLog("Legacy event flush error: $error.")
+                    enqueue {
+                        if (shouldMarkAsSynced) {
+                            legacyEventsFileHelper.clear(storedLegacyEventsWithNullValues.size)
+                        }
+                    }
+                },
+            )
+        }
+    }
+
+    private fun getStoredEvents(): List<BackendEvent?> {
+        var events: List<BackendEvent?> = emptyList()
+        fileHelper.readFile { sequence ->
+            events = sequence.take(flushCount).toList()
+        }
+        return events
+    }
+
+    private fun getLegacyStoredEvents(): List<BackendEvent> {
+        var events: List<PaywallStoredEvent?> = emptyList()
+        legacyEventsFileHelper.readFile { sequence ->
+            events = sequence.take(flushCount).toList()
+        }
+        return events
+            .filterNotNull()
+            .map { BackendEvent.Paywalls(it.toPaywallBackendEvent()) }
```

Comment:
Since this is returning `BackendEvent.Paywalls`, I would make that clear in the function name. Also you can use `mapNotNull` here and remove the `filterNotNull`